discord/app_commands/models.py

```diff
@@ -509,20 +515,22 @@ class Argument:
         '_state',
     )
 
-    def __init__(self, *, parent: ApplicationCommandParent, data: ApplicationCommandOption, state=None):
-        self._state = state
-        self.parent = parent
+    def __init__(
+        self, *, parent: ApplicationCommandParent, data: ApplicationCommandOption, state: Optional[ConnectionState] = None
+    ) -> None:
+        self._state: Optional[ConnectionState] = state
+        self.parent: ApplicationCommandParent = parent
         self._from_data(data)
 
     def __repr__(self) -> str:
         return f'<{self.__class__.__name__} name={self.name!r} type={self.type!r} required={self.required}>'
 
-    def _from_data(self, data: ApplicationCommandOption):
+    def _from_data(self, data: ApplicationCommandOption) -> None:
         self.type: AppCommandOptionType = try_enum(AppCommandOptionType, data['type'])
         self.name: str = data['name']
         self.description: str = data['description']
         self.required: bool = data.get('required', False)
-        self.choices: List[Choice] = [Choice(name=d['name'], value=d['value']) for d in data.get('choices', [])]
+        self.choices: List[_ChoiceTypes] = [Choice(name=d['name'], value=d['value']) for d in data.get('choices', [])]
```

Comment:
Instead of this you can just do `Choice[Union[int, float, str]]`. The `_ChoiceTypes` seems kind of unnecessary as a result.